ts/Series/Networkgraph/NetworkgraphSeries.ts

```diff
@@ -193,23 +200,100 @@ class NetworkgraphSeries extends Series {
         NodesComposition.destroy.call(this);
     }
 
+    /**
+     * Networkgraph needs a custom initDataLabels function
+     * to initiate the animation in a different way
+     * than for other series (drawDataLabels is called each frame)
+     * @private
+     */
+    public initDataLabels(this: NetworkgraphSeries): SVGElement {
+        const series = this,
+            dlOptions = series.options.dataLabels,
+            shouldInitialize = series.shouldInitializeDataLabels;
+
+        // if it's the first drawDataLabels() call
+        // then we'll create the SVG group for the first time
+        // & setup the animation from opacity 0 to opacity 1
+        // which lasts based on style.transition property
+        if (shouldInitialize) {
+            // create the SVG element for the dataLabels group
+            const dataLabelsGroup = this.initDataLabelsGroup();
+
+            // apply the dataLabels.style.transition not only to the
+            // individual dataLabels but also to the entire group
+            // to play the animation when the group appears
+            if (dlOptions && dlOptions.style && dlOptions.style.transition) {
+                dataLabelsGroup.attr({
+                    style: `transition: ${dlOptions.style.transition} ease 0s;`
+                });
+            }
+
+            // initialize the opacity of the group to 0 (start of animation)
+            dataLabelsGroup.attr({ opacity: 0 });
+
+            const group = series.dataLabelsGroup;
+            if (group) {
+                if (series.visible) { // #2597, #3023, #3024
+                    dataLabelsGroup.show();
+                }
```

Comment:
Could be simplified to `if (series.visible ) { ... }` (the group and variable is available, right?)